src/middleware/auth.js

```diff
@@ -0,0 +1,16 @@
+import jwt from 'jsonwebtoken';
+
+export const verifyToken = (req, res, next) => {
+  // eslint-disable-next-line dot-notation
+  const token = req.headers['authorization'];
+  if (!token) {
+    res.status(401).send('Access Denied');
+  }
+  try {
+    const checkToken = jwt.verify(token, process.env.TOKEN_KEY);
+    req.user = checkToken;
```

Comment:
You already have the decoded token here, you can make use of it to fix the comment I added in the getAgentProperty controller.